consensus/safety-rules/src/safety_rules.rs

```diff
@@ -108,7 +140,28 @@ impl<T: Payload> SafetyRules<T> {
             .next_epoch_state()
             .cloned()
             .ok_or(Error::InvalidLedgerInfo)?;
+
+        // By definition, the signer key shall be the same as consensus_key
+        // in case we have an unmatched key, retrieve from the storage the appropriate key
+        match self.verify_key(&epoch_state) {
+            // TODO: should we re-initialize the validator??
+            Err(Error::NotInitialized) => {
+                info!("Validator signer uninitialized.");
+            }
+            // TODO: This happens when the validator account is not even in the set
```

Comment:
nit: for multi-line comments, it's common practice to use full stops and/or capitalization to mark separation between sentences (new lines can be unclear, for example, this comment and the one on line 145).